### DISCUSSION TITLE
Is typst app using CRDT for collaborative editing?

### DISCUSSION THREAD

**Discussion by drriguz** [typst/webapp-issues#436] [General]
Hi, 

just for curious, what kind of architecture is using in typst app? Since the app is not open source, and it works really well, so I'm thinking how it is implemented. According to my knowledge, using CRDT based collaborative editing algorithm(eg. Yjs) + WASM for pure local rendering would be a possible choice, which would be really fast and also allows concurrent editing.

**Comment by laurmaedje** [comment]
Yes it is YJS with a custom WebSocket-based server and WASM for on-device compilation.
